eq-author-api/utils/datastore.js

```diff
@@ -0,0 +1,97 @@
+const {
+  QuestionnaireModel,
+  QuestionnaireVersionsModel,
+} = require("../db/models/DynamoDB");
+const { omit, isEqual } = require("lodash");
+
+const createQuestionnaire = questionnaire => {
+  return new Promise((resolve, reject) => {
+    new QuestionnaireModel({
+      ...omit(questionnaire, "sections", "metadata"),
+    }).save({}, err => {
+      if (err) {
+        reject(err);
+      }
+      new QuestionnaireVersionsModel({
```

Comment:
To avoid the call back inside a call back you could turn these into two methods each wrapping the functions in a promise. You might even be able to do this at the import easily with `util.promisify` https://nodejs.org/dist/latest-v10.x/docs/api/util.html#util_util_promisify_original

Reply:
so like this?
```
return util.promisify(new QuestionnaireModel({
    ...omit(questionnaire, "sections", "metadata"),
  }).save({})).then(() => {
    util.promisify(new QuestionnaireVersionsModel({
      ...questionnaire,
      metadata: questionnaire.metadata,
      updatedAt: Date.now(),
    }).save()).then(() => {
      return questionnaire
    })
  })
```

Reply:
No, you'd normally wrap the function but as it is an instance method this might not be so easy.